--- a/Modules/posixmodule.c	2026-05-25 12:10:27.715181369 -0700
+++ b/Modules/posixmodule.c	2026-05-25 12:10:27.734181409 -0700
@@ -16617,23 +16617,35 @@
 #endif
 
 #if HAVE_DECL_RTLD_LAZY
+#ifdef RTLD_LAZY
     if (PyModule_AddIntMacro(m, RTLD_LAZY)) return -1;
 #endif
+#endif
 #if HAVE_DECL_RTLD_NOW
+#ifdef RTLD_NOW
     if (PyModule_AddIntMacro(m, RTLD_NOW)) return -1;
 #endif
+#endif
 #if HAVE_DECL_RTLD_GLOBAL
+#ifdef RTLD_GLOBAL
     if (PyModule_AddIntMacro(m, RTLD_GLOBAL)) return -1;
 #endif
+#endif
 #if HAVE_DECL_RTLD_LOCAL
+#ifdef RTLD_LOCAL
     if (PyModule_AddIntMacro(m, RTLD_LOCAL)) return -1;
 #endif
+#endif
 #if HAVE_DECL_RTLD_NODELETE
+#ifdef RTLD_NODELETE
     if (PyModule_AddIntMacro(m, RTLD_NODELETE)) return -1;
 #endif
+#endif
 #if HAVE_DECL_RTLD_NOLOAD
+#ifdef RTLD_NOLOAD
     if (PyModule_AddIntMacro(m, RTLD_NOLOAD)) return -1;
 #endif
+#endif
 #if HAVE_DECL_RTLD_DEEPBIND
     if (PyModule_AddIntMacro(m, RTLD_DEEPBIND)) return -1;
 #endif
